crates/typst/src/layout/inline/mod.rs

```diff
@@ -180,17 +181,19 @@ impl<'a> Preparation<'a> {
 }
 
 /// A segment of one or multiple collapsed children.
-#[derive(Debug, Copy, Clone)]
+#[derive(Debug, Clone)]
 enum Segment<'a> {
     /// One or multiple collapsed text or text-equivalent children. Stores how
     /// long the segment is (in bytes of the full text string).
     Text(usize),
     /// Horizontal spacing between other segments.
     Spacing(Spacing),
     /// A mathematical equation.
-    Equation(&'a EquationElem),
-    /// A box with arbitrary content.
-    Box(&'a BoxElem, bool),
+    Equation(&'a EquationElem, Vec<MathParItem>),
+    /// A box with arbitrary content and fixed width.
+    Box(&'a BoxElem, Frame),
+    /// A box with fractional width.
+    FracBox(&'a BoxElem, Fr),
```

Comment:
I think the fr-related changes can be reverted because the order in which these things are layouted doesn't matter, as long as it is fixed.